internal/db/sqltest/tests/iam/iam_grant.sql

```diff
@@ -0,0 +1,277 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: BUSL-1.1
+
+begin;
+select plan(27);
+select wtt_load('widgets', 'iam');
+
+-- insert canonical_grant with valid resource
+-- validate the resource is set to 'scope'
+prepare insert_grant_scope as
+  insert into iam_grant
+    (canonical_grant)
+  values
+    ('type=scope;others=stuff;');
+select lives_ok('insert_grant_scope');
+select is(
+  (select resource
+     from iam_grant
+    where canonical_grant = 'type=scope;others=stuff;'),
+  'scope',
+  'resource should be set to "scope" by set_resource() trigger'
+);
+
+-- insert canonical_grant with no type
+-- the insert should fail because the type is malformed
+prepare insert_malformed_grant as
+  insert into iam_grant
+    (canonical_grant)
+  values
+    ('no_type_at_all;');
+select throws_like(
+  'insert_malformed_grant',
+  'value for domain wt_canonical_grant violates check constraint "wt_canonical_grant_check"',
+  'inserting a grant that is malformed should fail'
+);
+
+-- insert canonical_grant with type=role,group
+-- validate the resource is set to 'role'
```

Comment:
This comment is incorrect, this test is validating that an error is thrown, not that the resource was set to `role`.